src/StringTools/InternableString.cs

```diff
@@ -0,0 +1,343 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.StringTools
+{
+    /// <summary>
+    /// Represents a string that can be converted to System.String with interning, i.e. by returning an existing string if it has been seen before
+    /// and is still tracked in the intern table.
+    /// </summary>
+    /// <remarks>
+    /// The structure is public because it's exposed via the <see cref="TryInternStringDelegate"/> callback. It's a mutable struct and should
+    /// generally not be used directly from outside of the library.
+    /// </remarks>
+    public ref struct InternableString
+    {
```

Comment:
I'm a little confused about when someone should use InternableString vs. SpanBasedStringBuilder. Since they have some similarities in their implementation, some of my comments there might also apply here.

Reply:
You're right, it was confusing. The only reason `InternableString` was exposed was to pass it to the intern callback. Now that the callback is gone, the type can be internal and not part of the public surface. Thanks!